### PR TITLE
Metric with no_grad

### PR DESCRIPTION
**Pull request by anmolsjoshi**
This is a pull request for this [issue](https://github.com/pytorch/ignite/issues/221)

- Added torch.no_grad() to iteration_completed in Metric class
- Added test_no_grad to test/ignite/metrics/test_metric.py

### CODE REVIEW

**Comment by vfdev-5** [comment]
Thanks @anmolsjoshi !
